api/src/neo4j/queries/search/modelSearch.js

```diff
@@ -7,6 +7,49 @@ import {
   getScore,
 } from 'neo4j/queries/search/helper';
 
+const fetchIds = async ({ component, term, model, version, limit }) => {
+  // The search term is used twice, once with exact match and once with
+  // fuzzy match. This seems to produce optimal results.
+  let statement = `
+CALL db.index.fulltext.queryNodes("fulltext", "${term} ${term}~")
+YIELD node, score
+WITH node, score, LABELS(node) as labelList
+OPTIONAL MATCH (node:${model}:${component})
+OPTIONAL MATCH (node)-[${version}]-(parentNode:${model}:${component})
+WHERE node:${model} OR parentNode:${model}
+WITH DISTINCT(
+	CASE
+                WHEN EXISTS(node.id) AND NOT apoc.coll.intersection(${JSON.stringify(
+                  CHILD_LABELS
+                )}, LABELS(node))
+                THEN { id: node.id, labels: labelList, score: score }
+		ELSE { id: parentNode.id, labels: LABELS(parentNode), score: score }
+	END
+) as r
+WHERE
+	any(r IN r.labels WHERE r="${model}")
+	AND any(r IN r.labels WHERE r="${component}")
+RETURN r
+`;
+
+  if (limit) {
+    statement += `
+LIMIT ${limit}
+`;
+  }
+
+  const results = await queryListResult(statement);
+
+  const uniqueIds = {};
+  for (let [pos, node] of Object.entries(results)) {
```

Comment:
It is idiomatic to not use any placeholder and simply ignore the values of an array you are not interested in. See: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment

```suggestion
  for (let [, node] of Object.entries(results)) {
```